src/mlpack/tests/softmax_regression_test.cpp

```diff
@@ -392,14 +388,13 @@ BOOST_AUTO_TEST_CASE(SoftmaxRegressionOptimizerTrainTest)
   for (size_t i = 500; i < 1000; ++i)
     labels[i] = size_t(1.0);
 
-  SoftmaxRegressionFunction srf(dataset, labels, 2, 0.01, true);
-  L_BFGS<SoftmaxRegressionFunction> lbfgs(srf);
-  SoftmaxRegression<> sr(lbfgs);
+  L_BFGS lbfgs;
+  SoftmaxRegression<> sr(dataset, labels, 2, lbfgs, 0.01, true);
 
+  L_BFGS lbfgs2;
   SoftmaxRegression<> sr2(dataset.n_rows, 2, true);
-  L_BFGS<SoftmaxRegressionFunction> lbfgs2(srf);
-  sr2.Parameters() = srf.GetInitialPoint();
-  sr2.Train(lbfgs2);
+  sr2.Lambda() = 0.01;
+  sr2.Train(dataset, labels, 2, lbfgs2);
```

Comment:
For this one, we can initialize the parameters in the same way, but then train as

```
sr.Train(dataset, labels, 2, lbfgs);
sr2.Train(dataset, labels, 2, lbfgs2);
```